pkg/controller/common/container/container.go

```diff
@@ -53,22 +61,42 @@ const (
 	JournalbeatImage      Image = "beats/journalbeat"
 	PacketbeatImage       Image = "beats/packetbeat"
 	AgentImage            Image = "beats/elastic-agent"
-	MapsImage             Image = "elastic-maps-service/elastic-maps-server-ubi8"
+	MapsImage             Image = "elastic-maps-service/elastic-maps-server"
 	LogstashImage         Image = "logstash/logstash"
 )
 
 // ImageRepository returns the full container image name by concatenating the current container registry and the image path with the given version.
-func ImageRepository(img Image, version string) string {
+// A UBI suffix (-ubi8 or -ubi suffix depending on the version) is appended to the image name for the maps image,
+// or any image if the operator is configured with --ubi-only.
+func ImageRepository(img Image, ver version.Version) string {
 	// replace repository if defined
 	image := img
 
 	if containerRepository != "" {
 		image = Image(fmt.Sprintf("%s/%s", containerRepository, img.Name()))
 	}
 
-	// don't double append suffix if already contained as e.g. the case for maps
-	if strings.HasSuffix(string(img), containerSuffix) {
-		return fmt.Sprintf("%s/%s:%s", containerRegistry, image, version)
+	suffix := ""
+	UBIOnly := containerSuffix == UBISuffix
```

Comment:
nit: not a strong opinion but it feels to me that `UBIOnly` is the boolean flag that should be set by `--ubi-only`.

Reply:
I renamed `UBIOnly` in `useUBISuffix` in dd2ff5a.